Create integration test suite and migrate appropriate tests from unit test suite #590

erangeles opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/softlayer/sl-ember-components/1204)
<!-- Reviewable:end -->

juwara0 commented:
Review status: 0 of 2 files reviewed at latest revision, 11 unresolved discussions.

---

_[tests/integration/components/sl-input-test.js, line 42 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVBERfRCXJhyecK5l:-K3RVBERfRCXJhyecK5m:1524093011) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L42)):_
We do not need to do this render or the assert below to check that control-label is not set. Just render the template with the property being set and verify that the class gets applied.

---

_[tests/integration/components/sl-input-test.js, line 56 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVO5sxCafQhlmsHhy:-K3RVO5sxCafQhlmsHhz:412670943) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L56)):_
Please break this into two assertions one verifying that there is a class control-label and the other one can be this one.

---

_[tests/integration/components/sl-input-test.js, line 74 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVj_xXsW7jObujN3a:-K3RVj_yBAy_GZFMpEyu:1085918049) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L74)):_
This test and the one below are duplicates. You only need one or the other.

---

_[tests/integration/components/sl-input-test.js, line 93 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RWeJgT3GN4fy34t8l:-K3RWeJgT3GN4fy34t8m:1497256062) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L93)):_
instead lets do this assertion as:
```
assert.strictEqual(
        this.$( '>:first-child' ).find( '.text-info' ).length,
        1,
        '"optional" property sets class "text-info"'
    );
```

---

_[tests/integration/components/sl-input-test.js, line 111 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RWQQ1zvdKUawHaHi4:-K3RWQQ1zvdKUawHaHi5:1476349311) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L111)):_
instead lets do this assertion as:
```
assert.strictEqual(
        this.$( '>:first-child' ).find( '.text-danger' ).length,
        1,
        '"required" property sets class "text-danger"'
    );
```

---

_[tests/integration/components/sl-input-test.js, line 163 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RXDDvbTKliPE0I3YQ:-K3RXDDvbTKliPE0I3YR:960719498) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L163)):_
instead lets do this assertion as:
```
assert.strictEqual(
        this.$( '>:first-child' ).find( '.help-block' ).length,
        1,
        '"helpText" property sets class "help-block"'
    );
```

---

_[tests/integration/components/sl-input-test.js, line 239 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3V5z_HCeJH4UTXpGz-:-K3V5z_IKaM2Ee9DM3ux:-863335293) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L239)):_
please remove the spaces after " id : " so that it is "id: "

---

_[tests/integration/components/sl-input-test.js, line 277 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3Rbm-BA2y_TENOQ0O2:-K3Rbm-BA2y_TENOQ0O3:-1570597842) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L277)):_
We do not need to do this render or the assert below to check that the property is not set. Just render the template with the property being set and verify that the property gets applied.

---

_[tests/unit/components/sl-input-test.js, line 76 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3V8vZRDnkB7DpaaAWv:-K3V8vZRDnkB7DpaaAWw:1809351285) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/unit/components/sl-input-test.js#L76)):_
Please revert this change... it should be:

const inputElement = this.$( 'input' ).get( 0 );

---

_[tests/unit/components/sl-input-test.js, line 78 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3V8kf1jYgjW1nk2PkR:-K3V8kf1jYgjW1nk2PkS:-1804512258) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/unit/components/sl-input-test.js#L78)):_
Please revert this change... it should be:

const events = jQueryData( inputElement, 'events' );

---

_[tests/unit/components/sl-input-test.js, line 364 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3VARiNdOvoKR44ZIOc:-K3VARiNdOvoKR44ZIOd:-2088720051) ([raw file](https://github.com/softlayer/sl-ember-components/blob/2dc2f31be6cdcac6987dccb07847f517ebb5a3ec/tests/unit/components/sl-input-test.js#L364)):_
This test needs to stay. Do not remove it.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)_
<!-- Sent from Reviewable.io -->

juwara0 commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)_
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 0 of 2 files reviewed at latest revision, 11 unresolved discussions.

---

*[tests/integration/components/sl-input-test.js, line 42 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVBERfRCXJhyecK5l:-K4Nku_Q2JBatfzHJe0o:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L42)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 56 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVO5sxCafQhlmsHhy:-K4Nkv-Cvf96FUkkpcZP:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L56)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 74 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVj_xXsW7jObujN3a:-K4NkxIHSuLW7-jvmTdr:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L74)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 93 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RWeJgT3GN4fy34t8l:-K4Nm9c-gxb-26NanTJl:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L93)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 163 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RXDDvbTKliPE0I3YQ:-K4NmaBVqXlNABpTUtX2:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L163)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 239 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3V5z_HCeJH4UTXpGz-:-K4NmnVBsuIoUKEtI_7_:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L239)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 277 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3Rbm-BA2y_TENOQ0O2:-K4NmyVDx6fR7tyKW5Ju:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L277)):*
Done.

---

*[tests/unit/components/sl-input-test.js, line 76 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3V8vZRDnkB7DpaaAWv:-K4NnU8SDs8yTwZdx65p:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/unit/components/sl-input-test.js#L76)):*
Done.

---

*[tests/unit/components/sl-input-test.js, line 78 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3V8kf1jYgjW1nk2PkR:-K4NnUXpyVs-KoFB4l-r:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/unit/components/sl-input-test.js#L78)):*
Done.

---

*[tests/unit/components/sl-input-test.js, line 364 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3VARiNdOvoKR44ZIOc:-K4Nnfc7FZt6tt5TOPTZ:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/2dc2f31be6cdcac6987dccb07847f517ebb5a3ec/tests/unit/components/sl-input-test.js#L364)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

juwara0 commented:
@notmessenger: this issue is ready for your review

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-:-K4O91hXJ0tQJCtlIIes:1055644227)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
*[tests/integration/components/sl-input-test.js, line 42 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVBERfRCXJhyecK5l:-K4Oi96K1JYEgF2F3dvs:2113222824) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L42)):*
If this is "done" why is this render and assertion still in the PR?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/integration/components/sl-input-test.js, line 42 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K3RVBERfRCXJhyecK5l:-K4OnOpLl7qrTGbP041l:133607605) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b7c071f25965a0fcbe065a0b8422f187ea3510e7/tests/integration/components/sl-input-test.js#L42)):*
I removed the old render and assertion which was a "negative assertion". thus addressing michael's comment as "done".

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[tests/integration/components/sl-input-test.js, line 46 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YoizBsVGyzyF7l7C1:-K4YoizBsVGyzyF7l7C2:206756630) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L46)):*
use `assert.ok()` since `hasClass()` returns a boolean

---

*[tests/integration/components/sl-input-test.js, line 194 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YreKCJY5Cje6CwX1r:-K4YreKCJY5Cje6CwX1s:762505026) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L194)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

---

*[tests/integration/components/sl-input-test.js, line 199 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBDAqChh78W3W-qP:-K4YxBDAqChh78W3W-qQ:762505026) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L199)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

---

*[tests/integration/components/sl-input-test.js, line 204 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBxvc19k4pjzA5bl:-K4YxBxvc19k4pjzA5bm:762505026) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L204)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

---

*[tests/integration/components/sl-input-test.js, line 209 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxChKtLi5O90t_jRF:-K4YxChKtLi5O90t_jRG:762505026) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L209)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

---

*[tests/integration/components/sl-input-test.js, line 214 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxDMxI-1sg4VEzJJX:-K4YxDMxI-1sg4VEzJJY:762505026) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L214)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

---

*[tests/integration/components/sl-input-test.js, line 227 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxaPw2ycfvO1oGxR8:-K4YxaPw2ycfvO1oGxR9:-779987682) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L227)):*
What impact does this have on the test assertions?  I do not see anything tested or affected by this property.

---

*[tests/integration/components/sl-input-test.js, line 233 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxFSKEIda5aVdyVKY:-K4YxFSKEIda5aVdyVKZ:-1803954970) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L233)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

This seems to be the same assertion in the "Typeahead classes are present" test case.  What sets it apart/why is  it here also?

---

*[tests/integration/components/sl-input-test.js, line 238 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxXfSpJInXRoh2rfj:-K4YxXfSpJInXRoh2rfk:-1803954970) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L238)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

This seems to be the same assertion in the "Typeahead classes are present" test case.  What sets it apart/why is  it here also?

---

*[tests/integration/components/sl-input-test.js, line 243 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxYSJNA6nvBXVvJ1s:-K4YxYSJNA6nvBXVvJ1t:-1803954970) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L243)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

This seems to be the same assertion in the "Typeahead classes are present" test case.  What sets it apart/why is  it here also?

---

*[tests/integration/components/sl-input-test.js, line 248 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxZ9XAWIFG9dpKeen:-K4YxZ9XAWIFG9dpKeeo:-1803954970) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L248)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

This seems to be the same assertion in the "Typeahead classes are present" test case.  What sets it apart/why is  it here also?

---

*[tests/integration/components/sl-input-test.js, line 253 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxZt6I30PmwbSrnGR:-K4YxZt6I30PmwbSrnGS:-1803954970) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L253)):*
Needs to be an `assert.strictEqual()` against the `length()` of `find()`, as no results returned from `find()` will still be a jQuery object.

This seems to be the same assertion in the "Typeahead classes are present" test case.  What sets it apart/why is  it here also?

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K4Z-qHh_F4pAGyYvMWg:749533678) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
The value of "value" is set based on this logic:

```
const value = 'object' === Ember.typeOf( item ) ?
                            Ember.get( item, namePath ) : item;
```

not all of this logic is tested in this test case and should be.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
*[tests/unit/components/sl-input-test.js, line 77 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z6i6Bihi0D2CFXy-G:-K4Z6i6Bihi0D2CFXy-H:858048186) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L77)):*
Should not be referencing `_data` directly.  See https://github.com/softlayer/sl-ember-components/issues/917 and/or speak to @SpikedKira for more info

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 0 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[tests/integration/components/sl-input-test.js, line 46 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YoizBsVGyzyF7l7C1:-K4Z8_V5m9SgUOC6ShfS:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L46)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 194 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YreKCJY5Cje6CwX1r:-K4Z98N4k855XoGXhNXN:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L194)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 199 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBDAqChh78W3W-qP:-K4Z98iUjWAkT7zBLCh2:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L199)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 204 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBxvc19k4pjzA5bl:-K4Z994ktyCwGPuDRfUM:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L204)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 209 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxChKtLi5O90t_jRF:-K4Z99Mz-_AfMwvj58My:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L209)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 214 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxDMxI-1sg4VEzJJX:-K4Z99iFOhpH1AyI9z3P:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L214)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 227 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxaPw2ycfvO1oGxR8:-K4xxtKCKy_ZLdigvnII:-2071926540) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L227)):*
removed test and made a skip test for "suggestionsNamePath" prop.

---

*[tests/integration/components/sl-input-test.js, line 233 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxFSKEIda5aVdyVKY:-K4xy2sy3xTEq7zJlTM4:-2071926540) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L233)):*
removed test and made a skip test for "suggestionsNamePath" prop.

---

*[tests/integration/components/sl-input-test.js, line 238 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxXfSpJInXRoh2rfj:-K4xy3KTCoTpjTW8qRud:-2071926540) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L238)):*
removed test and made a skip test for "suggestionsNamePath" prop.

---

*[tests/integration/components/sl-input-test.js, line 243 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxYSJNA6nvBXVvJ1s:-K4xy3gGhlt8Qvq2WoNb:-2071926540) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L243)):*
removed test and made a skip test for "suggestionsNamePath" prop.

---

*[tests/integration/components/sl-input-test.js, line 248 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxZ9XAWIFG9dpKeen:-K4xy43ptE-5t2T1s_cm:-2071926540) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L248)):*
removed test and made a skip test for "suggestionsNamePath" prop.

---

*[tests/integration/components/sl-input-test.js, line 253 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxZt6I30PmwbSrnGR:-K4xy4cvwskYyP9C90Tz:-2071926540) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L253)):*
removed test and made a skip test for "suggestionsNamePath" prop.

---

*[tests/unit/components/sl-input-test.js, line 77 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z6i6Bihi0D2CFXy-G:-K4xxJMv2vCTSIju9cUz:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L77)):*
Done.

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K4xrf1EEX5vYXXqGWQw:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
tests/dummy/app/templates/demos/sl-input.hbs (with impact on the comment left for tests/dummy/app/controllers/demos/sl-input.js):  I see that you want to demonstrate this other way of specifying content, but you have done so at the expense of the "simpler" way.  Demonstrate both of them.

---

Reviewed 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[tests/dummy/app/controllers/demos/sl-input.js, line 4 \[r3\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4yGZxBOOGcMOUemY-U:-K4yGZxBOOGcMOUemY-V:-1298762889) ([raw file](https://github.com/softlayer/sl-ember-components/blob/a13565a8ce105004935d8b6b3249921f3e6fc7f9/tests/dummy/app/controllers/demos/sl-input.js#L4)):*
All of the same value option that were present before need to remain.  This is to demo that "bla" filters past "blue", for example, etc.

---

*[tests/integration/components/sl-input-test.js, line 194 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YreKCJY5Cje6CwX1r:-K4yDevc3ykJ6ydkENum:-598815112) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L194)):*
This assertion should be an `assert.strictEqual()` against the `length()` of `find()`.  There are two reasons for this:

* It aligns with all other similar occurrences in the rest of the code base
* It does not tie us to the structure of the DOM.  We are not concerned that there is a `span` but rather that there is an element with the class of `twitter-typeahead` applied to it.  If you wish to recommend the approach you have taken the we can do so but as a separate development effort, as it will affect more files than just the ones represented in this PR.

---

*[tests/integration/components/sl-input-test.js, line 199 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBDAqChh78W3W-qP:-K4yELNARoLmrE5j21ax:-811635599) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L199)):*
This assertion should be an `assert.strictEqual()` against the `length()` of `find()`.  The reasons for doing so are the same listed for the "twitter-typeahead class exists" assertion.

---

*[tests/integration/components/sl-input-test.js, line 204 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBxvc19k4pjzA5bl:-K4yE_09Nd_zJlj-ZZHJ:-811635599) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L204)):*
This assertion should be an `assert.strictEqual()` against the `length()` of `find()`.  The reasons for doing so are the same listed for the "twitter-typeahead class exists" assertion.

---

*[tests/integration/components/sl-input-test.js, line 209 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxChKtLi5O90t_jRF:-K4yEaDQDj8ev9f2lg3g:-811635599) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L209)):*
This assertion should be an `assert.strictEqual()` against the `length()` of `find()`.  The reasons for doing so are the same listed for the "twitter-typeahead class exists" assertion.

---

*[tests/integration/components/sl-input-test.js, line 214 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxDMxI-1sg4VEzJJX:-K4yEbL1j-c6Ic0eeCYv:-811635599) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L214)):*
This assertion should be an `assert.strictEqual()` against the `length()` of `find()`.  The reasons for doing so are the same listed for the "twitter-typeahead class exists" assertion.

---

*[tests/integration/components/sl-input-test.js, line 323 \[r3\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4yFFotD0UL6SFkTXUA:-K4yFFotD0UL6SFkTXUB:-1666749308) ([raw file](https://github.com/softlayer/sl-ember-components/blob/a13565a8ce105004935d8b6b3249921f3e6fc7f9/tests/integration/components/sl-input-test.js#L323)):*
Create a new issue regarding addressing these skipped tests

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K4yG6-kd1-hLJnFJCDr:-1534150388) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
The code you took out in this modification should have also remained in order to test both logic paths of the ternary assignment of `value`.

---

*[tests/unit/components/sl-input-test.js, line 221 \[r3\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4yFYK7d9yv3hke1KfI:-K4yFYK7d9yv3hke1KfJ:-1666749308) ([raw file](https://github.com/softlayer/sl-ember-components/blob/a13565a8ce105004935d8b6b3249921f3e6fc7f9/tests/unit/components/sl-input-test.js#L221)):*
Create a new issue regarding addressing these skipped tests

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-:-K4yH-GASwSHKSi9s0oc:893412359)*
<!-- Sent from Reviewable.io -->

erangeles commented:
@notmessenger added both examples.

---

Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[tests/dummy/app/controllers/demos/sl-input.js, line 4 \[r3\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4yGZxBOOGcMOUemY-U:-K4y_m27htxQ5y6p8pk2:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/a13565a8ce105004935d8b6b3249921f3e6fc7f9/tests/dummy/app/controllers/demos/sl-input.js#L4)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 194 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YreKCJY5Cje6CwX1r:-K4ycUzyJSOB4B4JjjmZ:-1620474753) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L194)):*
fixed.

---

*[tests/integration/components/sl-input-test.js, line 199 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBDAqChh78W3W-qP:-K4yc_NE-6pdu1ntw42r:336500263) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L199)):*
fixed. fyi: typeahead has 2 input fields with the class "typeahead". 

---

*[tests/integration/components/sl-input-test.js, line 204 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxBxvc19k4pjzA5bl:-K4yd5m9l26A1CPjR4yH:-1620474753) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L204)):*
fixed.

---

*[tests/integration/components/sl-input-test.js, line 209 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxChKtLi5O90t_jRF:-K4yd7S0OSsMB0yyIQWx:-1620474753) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L209)):*
fixed.

---

*[tests/integration/components/sl-input-test.js, line 214 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4YxDMxI-1sg4VEzJJX:-K4yd8H1dmbDKNk00p0A:-1620474753) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/integration/components/sl-input-test.js#L214)):*
fixed.

---

*[tests/integration/components/sl-input-test.js, line 323 \[r3\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4yFFotD0UL6SFkTXUA:-K4yIqV6IX1uQ7rV0Cxa:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/a13565a8ce105004935d8b6b3249921f3e6fc7f9/tests/integration/components/sl-input-test.js#L323)):*
Done.

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K4yepZOWqrFMdj_wmIi:-2060167119) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
there is 2 inputs so that `this.$( 'input' )` wouldnt work. the reason its ```component.get( 'value' )``` is because the value is being set in https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/addon/components/sl-input.js#L174 once we select a suggestion. now in regards to the ternary operator, in this case both https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/addon/components/sl-input.js#L171 get the key of the suggestion selected. so it would always be the same value thus only tested once. 

```javascript
<input id="ember396-input" type="text" class="form-control typeahead ember-view ember-text-field tt-input" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top; background-color: transparent;">
```

---

*[tests/unit/components/sl-input-test.js, line 221 \[r3\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4yFYK7d9yv3hke1KfI:-K4yIpmL8bqDM7hUGd0f:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/a13565a8ce105004935d8b6b3249921f3e6fc7f9/tests/unit/components/sl-input-test.js#L221)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-:-K4yY6jgPIuJFwTnedw5:1249175627)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 1 of 5 files reviewed at latest revision, 9 unresolved discussions.

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K4z1DQwX9i7mo3txCMP:240145945) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
disregard my last message, fixed. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Build is failing

---

Reviewed 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[tests/dummy/app/templates/demos/sl-input.hbs, line 34 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K50ln9xXvOe6cEU66MQ:-K50ln9xXvOe6cEU66MR:-359692746) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/dummy/app/templates/demos/sl-input.hbs#L34)):*
"names" ?

---

*[tests/dummy/app/templates/demos/sl-input.hbs, line 262 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K50py0LTDXctpw5noaS:-K50py0MB1FxAKAG3SDf:643431121) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/dummy/app/templates/demos/sl-input.hbs#L262)):*
Change to:

> Array of objects used as typeahead suggestions.  Each object must contain a key whose name matches the suggestionNamePath property value, which is "name" by default".

The "suggestionNamePath" text should be wrapped in `<code></code>` tags.

---

*[tests/integration/components/sl-input-test.js, line 195 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K511_y8qefzR8wOxrlw:-K511_y8qefzR8wOxrlx:-2050934833) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L195)):*
Remove "span" from `find()`.  We do not want to be tied to the structure of the DOM. We are not concerned that there is a span but rather that there is an element with the class of twitter-typeahead applied to it

---

*[tests/integration/components/sl-input-test.js, line 201 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512WIChp9hP-Y43sjF:-K512WIChp9hP-Y43sjG:1007449072) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L201)):*
Replace with:  `this.$( '>:first-child' ).find( '.typeahead' ).length`

---

*[tests/integration/components/sl-input-test.js, line 202 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512NexvFPWFtB5IOf-:-K512NexvFPWFtB5IOf0:1830890845) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L202)):*
Why are there two of these?  Is this due to how the plugin is rendering things?

---

*[tests/integration/components/sl-input-test.js, line 207 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512skPIRyO2V332OLA:-K512skPIRyO2V332OLB:-672308086) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L207)):*
Replace with `this.$( '>:first-child' ).find( '.tt-input' ).length,`

---

*[tests/integration/components/sl-input-test.js, line 213 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512zsZ83XBB9DD4P8N:-K512zsZ83XBB9DD4P8O:611125199) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L213)):*
Replace with `this.$( '>:first-child' ).find( '.tt-menu' ).length,`

---

*[tests/integration/components/sl-input-test.js, line 219 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K5134JhzMzKdx9p_Tbg:-K5134JhzMzKdx9p_Tbh:-1746032246) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L219)):*
Replace with ` this.$( '>:first-child' ).find( '.tt-dataset' ).length,`

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K513naMCS5ZHfoGXL4T:-38303101) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
I don't know what this test case should be doing by its description, but the conversation we are having should be that the existing test case of https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/tests/unit/components/sl-input-test.js#L353-L362 should remain, as well as both logic paths of the ternary should be tested.  I don't know how these have anything to do with a trigger?  Maybe it's just the name, as it's the trigger that then checks all this other data?  That's why I like the https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/tests/unit/components/sl-input-test.js#L353-L362 test case.  Seems perhaps these should remain separate.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-:-K50rvswTYxfALQy73Mz:618820389)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 2 of 5 files reviewed at latest revision, 10 unresolved discussions.

---

*[tests/dummy/app/templates/demos/sl-input.hbs, line 34 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K50ln9xXvOe6cEU66MQ:-K51FPrK1uqOw9V-1Z_Y:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/dummy/app/templates/demos/sl-input.hbs#L34)):*
Done.

---

*[tests/dummy/app/templates/demos/sl-input.hbs, line 262 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K50py0LTDXctpw5noaS:-K51FPNiJDNsJl-RXuVi:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/dummy/app/templates/demos/sl-input.hbs#L262)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 195 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K511_y8qefzR8wOxrlw:-K51EVOJZ6dRqlPrU_ZX:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L195)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 201 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512WIChp9hP-Y43sjF:-K51EWA2UseosNca49rO:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L201)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 202 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512NexvFPWFtB5IOf-:-K51EWRTdTY4zGmhqxn2:1457033220) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L202)):*
so the input field and the typeahead field both have ```typeahead``` class applied. thats why there is 2.

---

*[tests/integration/components/sl-input-test.js, line 207 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512skPIRyO2V332OLA:-K51EsK5RwZLb3Pg45SI:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L207)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 213 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K512zsZ83XBB9DD4P8N:-K51EseTH1DLawwBAls6:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L213)):*
Done.

---

*[tests/integration/components/sl-input-test.js, line 219 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K5134JhzMzKdx9p_Tbg:-K51Et2LGz1jvCI9at08:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/db7f4c9fc30216c7366bf502cad492523bef2a03/tests/integration/components/sl-input-test.js#L219)):*
Done.

---

*[tests/unit/components/sl-input-test.js, line 183 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K4Z-qHh_F4pAGyYvMWf:-K51ZWnAWiRz8mmgpAym:-1959463388) ([raw file](https://github.com/softlayer/sl-ember-components/blob/6c00552b0860e1a5e36c15bc24c429ddb2d50524/tests/unit/components/sl-input-test.js#L183)):*
so the ternary logic paths have been tested(both providing the same value, a string, as designed). you are right that the test desc was wrong and it has been updated and separated.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

erangeles commented:
@notmessenger can you re-run this build? seems an error on travis end

notmessenger commented:
It's actually related to https://github.com/softlayer/sl-ember-components/issues/1248  I will get this reviewed and merged in.

---

Review status: 2 of 5 files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-:-K52-X20_RCN_D_tv6RT:1226714094)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
https://github.com/softlayer/sl-ember-components/issues/1248 has been merged into the EmberCli-1.13.8 branch

---

Review status: 2 of 5 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-:-K522-i0LYREqKjX5cs-:-1297387044)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[tests/dummy/app/templates/demos/sl-input.hbs, line 262 \[r6\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204#-K522BGUjeo15JikL-8e:-K522BGUjeo15JikL-8f:834371545) ([raw file](https://github.com/softlayer/sl-ember-components/blob/b806b89f726a5cf5ba0375c2711d16efc0453549/tests/dummy/app/templates/demos/sl-input.hbs#L262)):*
Remove the extraneous `"` after *default*

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1204)*
<!-- Sent from Reviewable.io -->
